synthetics/datadog/datadog.go

```diff
@@ -0,0 +1,201 @@
+	for _, n := range slice {
+		if val == n {
+			return true
+		}
+	}
+	return false
+}
+
+// ContainsColon return whether a slice contains a specific value before the first colon
+func ContainsColon(slice []string, val string) bool {
+	for _, n := range slice {
+		if strings.ContainsAny(n, ":") {
+			s := strings.Split(n, ":")
+			n = s[0]
+		}
+		if val == n {
+			return true
+		}
+	}
+	return false
+}
+
+// createManagedSyntheticsTest configures and create a new synthetics test in Datadog
+func (tm *TestManager) createManagedSyntheticsTest(endpoint string, port int) (*datadog.SyntheticsTest, error) {
+	newOptions := &datadog.SyntheticsOptions{}
+	newOptions.SetAcceptSelfSigned(false)
+	newOptions.SetTickEvery(tm.CheckInterval)
+
+	expiryAssertion := &datadog.SyntheticsAssertion{}
+	expiryAssertion.SetType("certificate")
+	expiryAssertion.SetOperator("isInMoreThan")
+	expiryAssertion.Target = 12
+
+	newRequest := &datadog.SyntheticsRequest{}
+	newRequest.SetHost(endpoint)
+	newRequest.SetPort(port)
+
+	newConfig := &datadog.SyntheticsConfig{}
+	newConfig.Assertions = []datadog.SyntheticsAssertion{*expiryAssertion}
+	newConfig.SetRequest(*newRequest)
+
+	tags := tm.Tags
+	tags = append(tags, tm.DefaultTag)
+
+	newTest := &datadog.SyntheticsTest{Locations: []string{"aws:ap-northeast-1"}, Tags: tags}
+	newTest.SetName(endpoint)
+	newTest.SetType("api")
+	newTest.SetSubtype("ssl")
+	newTest.SetConfig(*newConfig)
+
+	newTest.SetMessage(tm.AlertMessage)
+	newTest.SetOptions(*newOptions)
+
+	test, err := tm.Client.CreateSyntheticsTest(newTest)
+	if err != nil {
+		return nil, err
+	}
+	return test, nil
+}
+
+// CreateManagedSyntheticsTests creates synthetics test according to the endpointList provided
+func (tm *TestManager) CreateManagedSyntheticsTests(endpointList []string) error {
+	if tm.DefaultTag == "" {
+		err := fmt.Errorf("No default tag is set for synthetics tests, aborting creation process")
+		return err
+	}
+	// Get all existing synthetic tests
+	tests, err := tm.Client.GetSyntheticsTests()
+	if err != nil {
+		log.Printf("Couldn't get synthetics tests from Datadog: %s\n", err.Error())
+		return err
+	}
+	for _, endpoint := range endpointList {
+		var matched bool
+		port := 443
+		if strings.ContainsAny(endpoint, ":") {
+			s := strings.Split(endpoint, ":")
+			endpoint = s[0]
+			i, err := strconv.Atoi(s[1])
+			if err != nil {
+				err := fmt.Errorf("The port number is not a valid numeral")
+				return err
+			}
+			port = i
+		}
+		// Normalize endpoint names from SYNTHETIC_ADDITIONAL_ENDPOINTS as they might have a defined port
+		for _, test := range tests {
+			// Only deal with tests having auto-generated tag
+			if Contains(test.Tags, tm.DefaultTag) {
+				endpointName, exists := test.GetNameOk()
+				// Use matched to determine whether a skip/create operation is required
+				if exists {
+					log.Printf("warn: endpointName (Test name) is %s, endpoint (Ingress name) is %s", endpointName, endpoint)
```

Comment:
Shouldn't this be after the `if endpoint == endpointName {` check, to only warn if they differ? In an `else` would be more readable, but the break technically makes it redundant.

```suggestion
					log.Printf("warn: endpointName (Test name) is %s, endpoint (Ingress name) is %s", endpointName, endpoint)
```